stdlib/public/core/FloatingPointParsing.swift.gyb

```diff
@@ -53,63 +53,89 @@ extension ${Self}: LosslessStringConvertible {
   /// Creates a new instance from the given string.
   ///
   /// The string passed as `text` can represent a real number in decimal or
-  /// hexadecimal format or special floating-point values for infinity and NaN
-  /// ("not a number").
+  /// hexadecimal format or can be in a special format representing infinity
+  /// or NaN ("not a number"). If the `text` is not in a recognized format,
+  /// the optional initializer will fail and return `nil`.
   ///
-  /// The given string may begin with a plus or minus sign character (`+` or
-  /// `-`). The allowed formats for each of these representations is then as
-  /// follows:
+  /// The `text` string may begin with a plus or minus sign character (`+` or
```

Comment:
I'm not sure if I like "may begin"; how does the following seem to you?

> The `text` string begins with an optional sign character (`+` or `-`) followed by one ...

Or maybe just mirror "consists of an optional plus or minus sign character" from the exponent section so that we're uniform?

Reply:
Done.